adapters/adquery/adquery.go

```diff
@@ -0,0 +1,211 @@
+func (a *adapter) buildRequest(bidReq *openrtb2.BidRequest, imp *openrtb2.Imp, ext *openrtb_ext.ImpExtAdQuery) *BidderRequest {
+	userId := ""
+	if bidReq.User != nil {
+		userId = bidReq.User.ID
+	}
+
+	return &BidderRequest{
+		V:                   prebidVersion,
+		PlacementCode:       ext.PlacementID,
+		AuctionId:           "",
+		BidType:             ext.Type,
+		AdUnitCode:          imp.TagID,
+		BidQid:              userId,
+		BidId:               fmt.Sprintf("%s%s", bidReq.ID, imp.ID),
+		Bidder:              bidderName,
+		BidderRequestId:     bidReq.ID,
+		BidRequestsCount:    1,
+		BidderRequestsCount: 1,
+		Sizes:               getImpSizes(imp),
+	}
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	if len(request.Imp) == 0 {
+		return nil, nil
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("x-openrtb-version", "2.5")
+
+	var result []*adapters.RequestData
+	var errs []error
+	for _, imp := range request.Imp {
+		ext, err := parseExt(imp.Ext)
+		if err != nil {
+			errs = append(errs, &errortypes.BadInput{err.Error()})
+			continue
+		}
+
+		requestJSON, err := json.Marshal(a.buildRequest(request, &imp, ext))
+		if err != nil {
+			return nil, []error{err}
+		}
+
+		data := &adapters.RequestData{
+			Method:  "POST",
+			Uri:     a.endpoint,
+			Body:    requestJSON,
+			Headers: headers,
+		}
+		result = append(result, data)
+	}
+
+	return result, errs
+}
+
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if responseData.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if responseData.StatusCode == http.StatusBadRequest {
+		err := &errortypes.BadInput{
+			Message: "Unexpected status code: 400. Bad request from publisher. Run with request.debug = 1 for more info.",
+		}
+		return nil, []error{err}
+	}
+
+	if responseData.StatusCode != http.StatusOK {
+		err := &errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info.", responseData.StatusCode),
+		}
+		return nil, []error{err}
+	}
+
+	respData, price, width, height, errs := parseResponseJson(responseData.Body)
+	if errs != nil && len(errs) > 0 {
+		return nil, errs
+	}
+
+	if respData == nil {
+		return adapters.NewBidderResponse(), nil
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(1)
+	if respData.Currency != "" {
+		bidResponse.Currency = respData.Currency
+	} else {
+		bidResponse.Currency = defaultCurrency
+	}
+	bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+		Bid: &openrtb2.Bid{
+			// There's much more possible fields to be added here, see OpenRTB docs for reference (type: Bid)
+			ID:      respData.ReqID,
+			ImpID:   request.Imp[0].ID,
```

Comment:
`request.Imp[0].ID` - only first imp is considered here. Do bidder server only intend to bid for first imp from request? If this is the expected behavior then should mention it in bidder docs

Reply:
The logic has been changed, now it supports multiple bids